docs/account.md

```diff
@@ -1,5 +1,219 @@
+
+## The types
+
+In live-common, there are currently 3 types of accounts that are defined in [types/account](../src/types/account.js):
+
+- `Account` which is a top level account associated to a `CryptoCurrency`.
+- `TokenAccount` which is a nested level account, **inside** an Account and that is associated to a `TokenCurrency`.
+- `ChildAccount` which is a variant of TokenAccount but more in context of a `CryptoCurrency` (typically used for Tezos)
+
+Like for [Currency](./currency.md), we have also a discriminator `.type` field.
+
+### Account
+
+```js
+type Account = {
+  type: "Account",
+  id: string,
+  seedIdentifier: string,
+  xpub?: string,
+  derivationMode: DerivationMode,
+  index: number,
+  freshAddress: string,
+  freshAddressPath: string,
+  freshAddresses: Address[],
+  name: string,
+  balance: BigNumber,
+  spendableBalance: BigNumber,
+  blockHeight: number,
+  currency: CryptoCurrency,
+  unit: Unit,
+  operationsCount: number,
+  operations: Operation[],
+  pendingOperations: Operation[],
+  lastSyncDate: Date,
+  subAccounts?: SubAccount[],
+  balanceHistory?: BalanceHistoryMap
+};
+```
+
+- `id` is a unique account identifier we build up with many informations. It's generally composed of 5 parts split by a `:` with:
+  - `implementation` the implementation the account is using (typically `js` or `libcore`). It would directly map the account to use the corresponding [AccountBridge](./AccountBridge.md).
+  - `version` a version number we have to allow for a migration system (changing it usually will force a recalculation of operations).
+  - `currencyId` tracks the unique id of currency.
+  - `xpubOrAddress` is the "restore key" which is either a xpub or an address. basically a way (combined with some other fields) to restore the account informations.
+  - `derivationMode` will be one of the derivation modes associated to the currency. See [derivation documentation](./derivation.md).
+- `seedIdentifier` is a unique way to identify a seed the account was associated with. The important criteria is it MUST be different between 2 seeds but it SHOULD be the same within siblings accounts inside a given `derivationMode`. We usually will use, in context of BIP44, the pubKey at `purpose'/coinType'`. For other accounts that don't uses BIP44, we have simply used the account address.
+- `xpub` is the xpub if relevant. This information is redundant with the `id` and might be eventually dropped.
+- `derivationMode` identify the derivation scheme that is used. See [derivation documentation](./derivation.md). example of values: `segwit`, `unsplit`, `native_segwit`,...
+- `index` is the index of the account among a given `derivationMode`.
+- `freshAddress` is the "next" public address where a user should receive funds. In context of Bitcoin, the address will be "renewed" each time funds were received in order to allow some privacy. In other blockchains, it might never changes.
```

Comment:
```suggestion
- `freshAddress` is the "next" public address where a user should receive funds. In the context of Bitcoin, the address will be "renewed" each time funds were received in order to allow some privacy. In other blockchains, it might never change.
```